kafkasql now uses the DB to obtain global Ids

EricWittmann opened the pull request:
OK this is a simple change to use the Kafka topic and the SQL database to generate global Ids in the KafkaSQL storage impl.  Next I will need to work on a simple import/export API to see how that works with this.

carlesarnal reviewed (approved):
LGTM, just a minor thing, the copyright value is 2020, I guess you'll need to update your IDE settings.

EricWittmann commented:
> 
> 
> LGTM, just a minor thing, the copyright value is 2020, I guess you'll need to update your IDE settings.

Fixed, thanks.  My IDE is ok, what happened is that I copy/pasted existing classes and modified the result, rather than created new classes.

EricWittmann commented:
@carlesarnal @famartinrh @jsenko This is now ready for review.  Fabian, this seems to have caused some testing errors, happy to work with you on those, along with designing a better test for import/export than what I have now.

EricWittmann commented:
Good catch, thanks @famartinrh .  All checks are green - what do you think is this good enough to merge?

famartinrh commented:
I'm working on the integration test for this, so as you prefer, merge now and I'll create a PR later with the test, or just wait :)

EricWittmann commented:
I'm happy to wait, just making sure you had eyeballs on this.  :)

famartinrh commented:
actually it's good to wait... I'm finding things that we may want to address now

the kafkasql storage is not waiting for all the import operations to be fullfilled. However the sql storage does so... wdyt? we may need to wait for all the submitted tasks

EricWittmann commented:
OK let me check.  Maybe we just need the `Reset` ID actions to be synchronous.

EricWittmann commented:
@famartinrh I made some changes to the kafkasql import implementation to make it synchronous.

EricWittmann commented:
Closing in favor of https://github.com/Apicurio/apicurio-registry/pull/1409
